internal/app/machined/pkg/controllers/k8s/kubelet_service.go

```diff
@@ -212,6 +216,92 @@ func (ctrl *KubeletServiceController) Run(ctx context.Context, r controller.Runt
 	}
 }
 
+// handlePolicyChange handles the cpuManagerPolicy change.
+func (ctrl *KubeletServiceController) handlePolicyChange(cfgSpec *k8s.KubeletSpecSpec, logger *zap.Logger) error {
+	const managerFilename = "/var/lib/kubelet/cpu_manager_state"
+
+	oldPolicy, err := loadPolicyFromFile(managerFilename)
+
+	switch {
+	case errors.Is(err, os.ErrNotExist):
+		return nil // no cpu_manager_state file, nothing to do
+	case err != nil:
+		return fmt.Errorf("error loading cpu_manager_state file: %w", err)
+	}
+
+	policy, err := getFromMap[string](cfgSpec.Config, "cpuManagerPolicy")
+	if err != nil {
+		return err
+	}
+
+	newPolicy := policy.ValueOrZero()
+	if equalPolicy(oldPolicy, newPolicy) {
+		return nil
+	}
+
+	logger.Info("cpuManagerPolicy changed", zap.String("old", oldPolicy), zap.String("new", newPolicy))
+
+	_, err = os.Stat(managerFilename)
```

Comment:
do we need to stat once again, given that we return on line 226?

Reply:
No, we do not, it's a leftover. Thanks for noticing!